File: Library/Portable/Control/GridLayout.cs
using System;
using System.Collections.Generic;
using System.Linq;
using MoonSharp.Interpreter;

namespace GwenNetLua.Control
{
	public class GridLayout : Internal.ControlBase
	{
		public static GridLayout Create(Internal.ControlBase parent)
		{
			return new GridLayout(new Gwen.Control.Layout.GridLayout(parent.Target));
		}

		[MoonSharpHidden]
		public GridLayout(Gwen.Control.ControlBase control)
			: base(control)
		{
		}

		public const float AutoSize = Gwen.Control.Layout.GridLayout.AutoSize;
		public const float Fill = Gwen.Control.Layout.GridLayout.Fill;

		public int ColumnCount { get { return GetTarget<Gwen.Control.Layout.GridLayout>().ColumnCount; } set { GetTarget<Gwen.Control.Layout.GridLayout>().ColumnCount = value; } }

		public void SetColumnWidths(IEnumerable<float> widths)
		{
			GetTarget<Gwen.Control.Layout.GridLayout>().SetColumnWidths(widths.ToArray());
		}

		public void SetRowHeights(IEnumerable<float> heights)
		{
			GetTarget<Gwen.Control.Layout.GridLayout>().SetColumnWidths(heights.ToArray());
		}
	}
}

File: Sample/Sample/Lua/Items/GridLayout.lua

local function CreateControl(parent, text)
	button = Gwen.Button.Create(parent)
	button.Text = text
end

this = Gwen.DockLayout.Create(Gwen.Canvas)

	grid = Gwen.GridLayout.Create(this)
	grid.Dock = Gwen.Dock.Fill
	grid.SetColumnWidths({ 0.2, Gwen.GridLayout.AutoSize, 140.0, 0.8 })
	grid.SetRowHeights({ 0.2, Gwen.GridLayout.AutoSize, 140.0, 0.8 })

	CreateControl(grid, "C: 20%, R: 20%")
	CreateControl(grid, "C: Auto R: 20%")
	CreateControl(grid, "C: 140, R: 20%")
	CreateControl(grid, "C: 80%, R: 20%")

	CreateControl(grid, "C: 20%, R: Auto")
	CreateControl(grid, "C: Auto R: Auto")
	CreateControl(grid, "C: 140, R: Auto")
	CreateControl(grid, "C: 80%, R: Auto")

	CreateControl(grid, "C: 20%, R: 140")
	CreateControl(grid, "C: Auto R: 140")
	CreateControl(grid, "C: 140, R: 140")
	CreateControl(grid, "C: 80%, R: 140")

	CreateControl(grid, "C: 20%, R: 80%")
	CreateControl(grid, "C: Auto R: 80%")
	CreateControl(grid, "C: 140, R: 80%")
	CreateControl(grid, "C: 80%, R: 80%")

AddItem("Layout", "GridLayout", this)
